### DISCUSSION TITLE
Enable sql syntax highlighting in multiline strings that start with a --sql comment

### DISCUSSION THREAD

**Discussion by saolof** [helix-editor/helix#8221] [Ideas]
A somewhat common way to identify a multiline string as an sql string for editor support is something like this:

```py
foo = """--sql
    select foo from bar
"""
```
(example: https://github.com/barklan/inline_sql_syntax )

It would be neat to have this or equivalent support for syntax highlighting in helix, which has otherwise been more or less perfect for me out of the box and has become my goto editor.

**Comment by the-mikedavis** [comment]
Injections are the right tool for this for languages with tree-sitter grammars: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection

Supporting this for plaintext is out of scope though, it should be limited to languages with tree-sitter grammars.

**Comment by danillos** [comment]
I did it for Ruby [here](https://github.com/helix-editor/helix/pull/6587/files#diff-170281381682e84da7908cc3a42f65c55a5ea8dc4f30b637c5c0d04eec4b1f93R4), it could help you to implement in other languages.

**Comment by johan-lejdung** [comment]
I did this to go simply by adding a file `~/.config/helix/runtime/queries/go/injections.scm` with the contents:
```
((
  (raw_string_literal) @constant
  (#match? @constant "(SELECT|select|insert|INSERT).*")
)@injection.content (#set! injection.language "sql"))
```

Extend to include as many keywords as you'd like :) Give it a go!

**Reply by noahfraiture** [reply]
Hey, does that still works for you on latest master ? It doesn't for me. I got this to work in python
```
(((string_content) @sql
                   (#match? @sql "^\\s*--sql")) @injection.content
 (#set! injection.language "sql"))
```
But it isn't really what I want and it's python, it doesn't work in go with raw_string_literal

Even this doesn't work `((raw_string_literal) @injection.content (#set! injection.language "sql"))` in go but some test with go-tree-sitter showed me that a string should indeed be `raw_string_literal`
